### PR TITLE
Bot model

### PR DESCRIPTION
**Pull request by tausifmuzaffar**
Creating models one at a time

https://drive.google.com/file/d/0B9BRb52RN_BFdnhRVm1Qd0phU1U/view?usp=sharing

### CODE REVIEW

File: db/model/botData.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * db/model/botData.js
+ *
+ * Bot webapps will have data that refocus will want to retrieve
+ * or store. This table will define the data information to be
+ * used manipulate the bots data.
+ */
```
Comment (kamilsmuga) [inline db/model/botData.js]: What are examples of such data? I'm having a hard time to find examples that would tie to the Bot model. If they exist - maybe should be part of the Bot model available as fields? 
All examples I can think of have a relationship with a Room model.

File: db/model/botData.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+const constants = require('../constants');
+
+const assoc = {};
+
+module.exports = function user(seq, dataTypes) {
```
Comment (kamilsmuga) [inline db/model/botData.js]: function name - user - is incorrect.
Side note - I don't think we need to name these functions at all - anonymous would do. However, other models seems to assign name to functions

File: db/helpers/botUtils.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
+/**
```
Comment (iamigo) [inline db/helpers/botUtils.js]: Needs additional tests as described here: https://coveralls.io/builds/11832728/source?filename=db%2Fhelpers%2FbotUtils.js

Comment (iamigo) [inline db/helpers/botUtils.js]: The nice thing about separating out these validation fns and exporting them as functions is that you can test them outside of the context of a running db.

File: db/model/bot.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * db/model/bot.js
+ *
+ * Bots are distributed webapps that will interact with our
+ * different routes. This object will be used to track
+ * and organize the bots that are associated with refocus.
```
Comment (iamigo) [inline db/model/bot.js]: Can you add a little more information here about bot actions and data, how those fields are populated and used?

**Comment by iamigo** [comment]
@tausifmuzaffar once we merge this, subsequent field changes must have migrations, too.

**Comment by tausifmuzaffar** [comment]
@iamigo @pallavi2209 thanks! Do field additions need migrations as well or just field changes?

**Comment by iamigo** [comment]
Just field changes

**Pull request merged**